lib/update.ml

```diff
@@ -993,6 +1000,28 @@ let reconstitute_date_dmy conf var =
   in
   (d, force_f_cal)
 
+let is_illegal_access_update old_access new_access =
+  match old_access, new_access with
+  | IfTitles, Public
+  | IfTitles, Private
+  | Public, Private
+  | Private, Public -> true
+  | _ -> false
+
+let check_illegal_access_update base person =
+  let old_access_opt = try
+      let iper = person.key_index in
+      let old_person = Gwdb.poi base iper in
+      Some (Gwdb.get_access old_person)
+    with _ -> None
```

Comment:
Could we avoid this *catch-all* `try … with …`?

Reply:
yes we can